### DISCUSSION TITLE
2.0 版本 中的 react types写法疑问

### DISCUSSION THREAD

**Discussion by 729993031** [alibaba/formily#1314] [Ideas]
101行中的Path<T[Key], Exclude<keyof T[Key], keyof Array<any>>> 是否能够修改为Path<T[Key]> 
为什么需要keyof  Array？

**Comment by liuweiGL** [comment]
你得看这个类型的作用：
![image](https://user-images.githubusercontent.com/21048878/116220684-7b860d00-a77f-11eb-957d-63861da56130.png)
可以看到 `Path` 是用来遍历 `object` 的 `key` 路径的，不把 `keyof any[]` 排除掉的话，如果存在 `array` 类型的值，那获取到的路径就错了：
![image](https://user-images.githubusercontent.com/21048878/116220912-aff9c900-a77f-11eb-9ddf-372806f74d7f.png)

**Comment by liuweiGL** [comment]
不得不说作者的体操做的很优秀 😄

**Reply by 729993031** [reply]
![image](https://user-images.githubusercontent.com/13050361/116228776-4d58fb00-a788-11eb-8f66-70a92689df81.png)

跟下面的这种写法有什么区别码？
